### DISCUSSION TITLE
Trying to setup talos cluster on HPE server with failed results

### DISCUSSION THREAD

**Discussion by matjazantloga** [siderolabs/talos#9169] [Q&A]
Hi, 

I've tried, together with teammate Sergey to setup talos cluster in different ways
- via local docker with success 
- vmware workstation with fail
-  govc on vcenter with fail
-  HPE server with fail
- 
This last example , HPE server is maybe simplest and i will just paste my config here and I hope i will get some clue what we are doing wrong

so, i boot my HPE gen9 dl 390 and got Talos up in maintenance mode with this IP 10.99.100.72 (example)

then i run following

**talosctl gen config TalosHPE https://10.99.100.190:6443**

and got **talosconfig** **controlplane.yaml** and **worker.yaml**

i modified **controlplane.yaml** with following information

   **network:
      interfaces:
        - interface: eth0
          dhcp: false
          addresses:
            - 10.99.100.191/24
          routes:
            - network: 0.0.0.0/0
              gateway: 10.99.100.1**

didn't change anything else

then i run 

**talosctl apply-config --insecure --nodes 10.99.100.72 --file controlplane.yaml**

end result in pic, before that talos was restarted and I clearly saw partitions were made etc ... 

![talos1](https://github.com/user-attachments/assets/87b91e4b-10b1-48c3-8882-f6270d6f547b)

**Comment by matjazantloga** [comment]
this post is embarrassment, i pull out some vmware ssd drives but forgot to put into small blank ssd, so it was installed nowhere ;D. Will post later I hope with more greens.

**Reply by matjazantloga** [reply]
![image](https://github.com/user-attachments/assets/b256f36a-cabb-4bba-aafc-2e4f4108f6f0)

**Comment by matjazantloga** [comment]
aaa, almost
![image](https://github.com/user-attachments/assets/80901818-0236-4336-8f6d-36afcb2542c6)
![image](https://github.com/user-attachments/assets/05ae5ffc-1701-41f8-98b6-99da6dd9029d)

**Reply by smira** [reply]
The machine in the screenshot above hasn't been installed, it doesn't have machine config. So the command fails as expected.

**Comment by matjazantloga** [comment]
this is command i will apply

talosctl apply-config --nodes 10.99.100.130 --endpoints 10.99.100.130  --file controlplane.yaml --talosconfig talosconfig --insecure

-----
talosconfig should be ok 
![image](https://github.com/user-attachments/assets/f8ce42f2-9b31-44c3-8372-d367c7065a7d)

most vital parts of controlplane.yaml, maybe i did sth wrong here
![image](https://github.com/user-attachments/assets/ef1fec65-1878-4e08-857f-11bcb89cb6be)
![image](https://github.com/user-attachments/assets/35c76711-894d-4857-acd3-0d5f00887ed7)
![image](https://github.com/user-attachments/assets/593e5e1a-a398-4991-a73b-31cff039d773)
![image](https://github.com/user-attachments/assets/65a4d5a3-cd34-46b0-a2bd-b5a07c7aafc0)
![image](https://github.com/user-attachments/assets/c61d1ca9-4fbc-4e86-a9a9-0e1d156a9f7b)
![image](https://github.com/user-attachments/assets/d0518c44-0cb8-46b8-9edf-0d4d12af9802)


![image](https://github.com/user-attachments/assets/3be09e30-82b4-44f0-b4cf-15abd22aaeb4)

**Comment by matjazantloga** [comment]
and it fails

![image](https://github.com/user-attachments/assets/6c7343a0-ca0d-4e3d-93bc-6418d2db9161)

**Comment by matjazantloga** [comment]
![image](https://github.com/user-attachments/assets/6a7c6fc4-1480-4533-a7f4-02c2afef58d0)
i still can ping the IP... also no idea why suddenly two GW are shown

**Comment by smira** [comment]
Probably you need to take a step back and try with a simplified configuration.

I think you configured networking both via machine config and via dashboard. Don't mix everything together, and try to get the machine to boot and get network access to it in maintenance mode, then apply the config.

**Reply by matjazantloga** [reply]
i don't understand why it gets static IP when i boot into reset maintenance mode, it should get IP from dhcp this was working but after i was doing some configs inside dashboard F3 console it doesn't work anymore, i would like to reset completely ...

also , i've defined /dev/sdc but i see its dealing with /dev/sda which is USB from which i'm loading Talos
i will paste screenshot for this

**Reply by matjazantloga** [reply]
![image](https://github.com/user-attachments/assets/2c02169c-5510-45c8-8bb2-3efa07ac3cd0)

**Reply by matjazantloga** [reply]
ok, so i reset network back to dhcp in maintenance mode and will try to apply config to that IP
![image](https://github.com/user-attachments/assets/c3c742f4-e8a1-436f-ba05-108ecdecd5a0)

**Comment by matjazantloga** [comment]
adding .82 into talosconfig
![image](https://github.com/user-attachments/assets/96028c87-5bd9-4296-8908-d727cdaddb69)

running with .82 and --insecure

![image](https://github.com/user-attachments/assets/536d7e17-25c8-473d-aa30-32836275cbc3)
![image](https://github.com/user-attachments/assets/c0056745-561e-42bd-8046-96bff94512b3)

.82 was removed but instead of adding .130, as can be seen in above controlplane.yaml config it adds .83 via dhcp and IP is then not visible and it also mess with sda instead of sdc.... i'm lost :(

**Reply by matjazantloga** [reply]
![image](https://github.com/user-attachments/assets/51cabcd8-9261-40d1-91cf-1e4b55e98024)
100.130 is applied, because this works.
just IP is not shown and for sure it's unhealthy 
![image](https://github.com/user-attachments/assets/14864838-4ba4-4770-b567-3d5aaaf0262b)

**Reply by matjazantloga** [reply]
gw 10.99.100.129 is no where to be found 

![image](https://github.com/user-attachments/assets/c31337d5-6fa0-400b-969d-745abb240df3)

alltough is configured in controlplane.yaml
prob sth wrong with network part

![image](https://github.com/user-attachments/assets/c338bd48-b658-44db-9990-6e18b14249fc)

**Reply by matjazantloga** [reply]
i just found out that network section of controlplane.yaml is not taken into apply config at all, i just made up some values
![image](https://github.com/user-attachments/assets/0cc00414-821b-4ed1-8a3c-0ea923379842)
and when i do --dry-run no error occurs
![image](https://github.com/user-attachments/assets/ddd51e77-5d09-4e71-b9bb-1a5b53182242)

**Reply by matjazantloga** [reply]
not really true, of course network config is taken into apply but what i found strange is this
when i have and for sure should have route config i got this

![image](https://github.com/user-attachments/assets/ce98dd81-24da-4848-9032-eaed0911e556)

![image](https://github.com/user-attachments/assets/e64b7f4a-afad-4afb-9e5b-3370dce35c89)

instead of adding this route its throw out error that route already exists

**Comment by matjazantloga** [comment]
sth is cooking :)

![image](https://github.com/user-attachments/assets/fcf8a854-d08f-4d85-a1a2-273480e83690)

**Comment by matjazantloga** [comment]
![image](https://github.com/user-attachments/assets/f6423e43-8173-46c6-818b-047734454944)
HPE node added to fresh cluster.
i0m happy :). Can't believe we lost few days bc our internal subnet is also 10.9x.xx.xx

**Comment by matjazantloga** [comment]
node IP was in conflict with pod and service subnets. we are using same range for our internal IPs.

once i changed those pod and service subnets out of our range everything started getting together.

[cid:b30338b4-7b2d-492b-8a11-73854138aa9f]


[cid:f50c4aab-d3d2-44af-9179-b20557ac41fd]

machine section is really simple, no need to complicate here.

HPE really has eno1 and /dev/sdb but it figured it out and was happy with above for config start


Best regards, s prijaznimi pozdravi

Matjaž Antloga, Ortoscale d.o.o.

________________________________

**Comment by miran248** [comment]
Thanks for the reply!
Might have the same issue. That or it's because i tried adding a new machine (with different secrets and config) to the existing omni cluster.
Existing cluster has 5 machines, all with public ipv4 and default config, new machine sits on a private network with a router in front (with public ip) and because of that i had to configure it differently.. Will make a brand new cluster today with all machines on the same setup and hopefully it will work :)
LP

**Reply by matjazantloga** [reply]
for sure it needs to be secrets from existing config, just edit machine part of controlplane or worker to suit the new machine and appy it
